---
title: "Investigate Chalc pop structure"
author: "Devon DeRaad"
date: "11/20/2020"
output: html_document
---

```{r}
library(RADstackshelpR)
library(vcfR)
library(adegenet)
library(ggplot2)
library(StAMPP)
library(ggtree)
library(scatterpie)

#read in vcf as vcfR
vcfR <- read.vcfR("~/Desktop/chalcophaps.rad/m3.vcf")

#subset the vcf to remove the 10 samples with problematic batch effects
vcfR@gt<-vcfR@gt[,colnames(vcfR@gt) != "Chalc.indi.WAM.22872" & colnames(vcfR@gt) != "Chalc.indi.AMNH.21441" & colnames(vcfR@gt) != "Chalc.indi.CAS.777" & colnames(vcfR@gt) != "Chalc.indi.ANWC.B55964" & colnames(vcfR@gt) != "Chalc.indi.AMNH.21534" 
               & colnames(vcfR@gt) != "Chalc.indi.AMNH.21572" & colnames(vcfR@gt) != "Chalco.step.KU.27832" & colnames(vcfR@gt) != "Chalco.step.KU.12228" & colnames(vcfR@gt) != "Chalco.step.WAM.26645" & colnames(vcfR@gt) != "Chalc.indi.AMNH.21615"
               & colnames(vcfR@gt) != "Chalco.step.KU.6905" & colnames(vcfR@gt) != "Chalco.step.KU.7287" & colnames(vcfR@gt) != "Chalc.indi.WAM.22720"& colnames(vcfR@gt) != "Chalco.indi.KU.31155"]

#read in sampling file
sampling<-read.csv("~/Desktop/chalcophaps.rad/chalc.sampling.csv")
#place java
sampling[35,c(7,8)]<- c(-7.2,111.2)
sampling[36,c(7,8)]<- c(-7.2,111.2)

#subset out the samples that weren't affected by batch effect
new.sampling<-sampling[sampling$id %in% colnames(vcfR@gt),]
#generate new pop
popmap<-data.frame(id=new.sampling$id,pop=new.sampling$Species)
```

execute hard filters
```{r}
#hard filter to minimum depth of 5, and minimum genotype quality of 30
vcfR<-hard.filter.vcf(vcfR=vcfR, depth = 3, gq = 30)

#execute allele balance filter
vcfR<-filter.allele.balance(vcfR)

#visualize and pick appropriate max depth cutoff
max_depth(vcfR)
#filter vcf by the max depth cutoff you chose
vcfR<-max_depth(vcfR, maxdepth = 200)

#check vcfR to see how many SNPs we have left
vcfR
dev.off()
```

```{r}
#missing by sample
miss<-missing.by.sample(vcfR=vcfR, popmap = popmap)
#run function to drop samples above the threshold we want from the vcf
#vcfR<-missing.by.sample(vcfR=vcfR, cutoff = .91)
```

```{r}
#visualize missing data by SNP and the effect of various cutoffs on the missingness of each sample
missing.by.snp(vcfR)

#choose a value that retains an acceptable amount of missing data in each sample
#and maximizes SNPs retained while minimizing overall missing data, and filter vcf
vcfR<-missing.by.snp(vcfR, cutoff = .75)
vcfR
```

```{r}
#PCA
gen<- vcfR2genlight(vcfR)
pca<-glPca(gen, nf=6)

#pull pca scores out of df
pca.scores<-as.data.frame(pca$scores)
pca.scores$species<-new.sampling$Species

#ggplot color by species
ggplot(pca.scores, aes(x=PC1, y=PC2, color=species)) +
  geom_point(cex = 2)
#ggplot color by species
ggplot(pca.scores, aes(x=PC3, y=PC4, color=species)) +
  geom_point(cex = 2)

```

```{r}
#calculate divergence btwn samples
gen@pop<-new.sampling$Species
inds<-stamppNeisD(gen, pop = FALSE)

#plot tree colored by subspecies
tree <- nj(inds)
plot(tree, type="unrooted", cex=.5)
ggtree(tree)+
  geom_tiplab(cex=1.8)
```

```{r}
#3 clusts
grp<-find.clusters(gen, n.pca=30, n.clust=3)
#set manually the values I chose from scree plots
dapc1<-dapc(gen, grp$grp, n.da = 5, n.pca =6)
compoplot(dapc1, legend=FALSE, show.lab =TRUE, cex.names=.4)

#4 clusts
grp<-find.clusters(gen, n.pca=30, n.clust=4)
#set manually the values I chose from scree plots
dapc1<-dapc(gen, grp$grp, n.da = 5, n.pca =6)
compoplot(dapc1, legend=FALSE, show.lab =TRUE, cex.names=.4)
```

```{r}
post<-as.data.frame(dapc1$posterior)
post$id<-rownames(post)
plot.pies<-merge(post, new.sampling, by= "id")
#
pac<-map_data("world")
ggplot()+
  geom_polygon(data = pac, aes(x=long, y = lat, group = group), fill="grey", col="black", cex=.1)+
  coord_sf(xlim = c(95, 170), ylim = c(-35, 30)) + 
  geom_scatterpie(aes(x=Long, y=Lat, group = Lat), 
                  data = plot.pies, cols = colnames(plot.pies[,c(2:5)]))+
  theme_classic()
```

```{r}
#make manual color vector
#col.vec<-(c(rep(4, times=11),rep(1, times=8),3,1,1,1,rep(2, times=4)))
col.vec<-(c(rep(1, times=11),rep(4, times=8),3,4,4,4,rep(2, times=4)))

gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
n = 4
cols = gg_color_hue(n)
j=1
#convert subspecies to colors
for (i in levels(as.factor(col.vec))){
  col.vec[col.vec == i]<-cols[j]
  j<-j+1
}

#plot tree colored by DAPC assignment
tree <- nj(inds)
ggtree(tree)+
  geom_tippoint(color=col.vec)+
  geom_tiplab(cex=1.8)
```
